src/mcs/montecarlo.jl

```diff
@@ -170,58 +188,98 @@ function clear_results(mcs::MonteCarloSimulation)
 the maximum defined time period otherswise. If `pretrial_func` or `posttrial_func` are defined,
 the designated functions are called just before or after (respectively) running the trial. The 
 functions must have the signature fn(m::Model, mcs::MonteCarloSimulation, trialnum::Int).
+
+If provided, `loop_args` must be a Vector of Pairs, where each pair is a symbol and a Vector
+of arbitrary values that will be meaningful to `loop_func`, which is called with the Model, 
+`m`, the MonteCarloSimulation `mcs`, and the splatted tuple of values for the current outer
+loop iteration.
 """
 function run_mcs(m::Model, mcs::MonteCarloSimulation, trials::Union{Vector{Int}, Range{Int}}; 
-                 ntimesteps=typemax(Int), output_dir=nothing, 
-                 pre_trial_func=nothing, post_trial_func=nothing)
+                 ntimesteps::Int=typemax(Int), 
+                 output_dir::Union{Void, AbstractString}=nothing, 
+                 pre_trial_func::Union{Void, Function}=nothing, 
+                 post_trial_func::Union{Void, Function}=nothing,
+                 loop_func::Union{Void, Function}=nothing,
+                 loop_args=nothing)
+
+    if (loop_func == nothing) != (loop_args == nothing)
+        error("run_mcs: loop_func and loop_arg must both be nothing or both set to non-nothing values")
+    end
+
     if m.mi == nothing
         build(m)
     end
+    md = m.mi.md
+    
+    orig_output_dir = output_dir
 
-    mcs.output_dir = output_dir
+    if loop_args == nothing
+        arg_tuples = (nothing,)     # handles case with no outer loop
+    else
+        seqs = [arg.second for arg in loop_args]
+        arg_tuples = product(seqs...)
+    end
 
-    clear_results(mcs)
+    for tup in arg_tuples
 
-    # Compute how often to print a "Running trial ..." message.
-    count = length(trials)
-    divisor = (count < 50 ? 1 : (count < 500 ? 10 : 100))
+        if loop_func != nothing
+            # Call outer loop setup function
+            loop_func(m, mcs, tup...)
 
-    for trialnum in trials
-        trialnum % divisor == 0 && println("Running trial $trialnum ")
-        _perturb_parameters(m, mcs, trialnum)
+            # Create a subdir to store the results of each outer loop
+            if orig_output_dir != nothing
+                output_dir = joinpath(orig_output_dir, join(map(string, tup), "_"))
+            end
+        end
 
-        pre_trial_func  != nothing && pre_trial_func(m, mcs, trialnum)
-        run(m, ntimesteps=ntimesteps)
-        post_trial_func != nothing && post_trial_func(m, mcs, trialnum)
+        clear_results(mcs)
+        
+        # Save the params to be perturbed so we can reset them after each trial
+        original_values = _copy_mcs_params(mcs, md)
 
-        store_trial_results(m, mcs, trialnum)
-    end
+        # Compute how often to print a "Running trial ..." message.
+        count = length(trials)
+        divisor = (count < 50 ? 1 : (count < 500 ? 10 : 100))
```

Comment:
We could just use https://github.com/timholy/ProgressMeter.jl here, that would probably result in nicer output.

Reply:
I created an issue for that.